Add React To A Website Translation

Anishpras opened the pull request:
<!--

Thank you for the PR! Contributors like you keep React awesome!

Please see the Contribution Guide for guidelines:

https://github.com/reactjs/reactjs.org/blob/main/CONTRIBUTING.md

If your PR references an existing issue, please add the issue number below

-->

Anishpras commented:
Hey, @arshadkazmi42 I have completed the translation, can you please review the changes.

arshadkazmi42 commented:
Thanks @Anishpras 

I will get to is ASAP.

Anishpras commented:
Thankyou!

Anishpras commented:
> Thanks @Anishpras
> 
> I will get to is ASAP.

@arshadkazmi42 Can I translate other pages while you review or should I wait for this PR to be merged?

arshadkazmi42 commented:
@Anishpras 

Lets get this PR merged first, as sometimes we get overwhelmed by too many pending PRs :P

Anishpras commented:
@arshadkazmi42 Okay sure. Thank you 😊.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
 ---
-title: Add React to a Website
+title: किसी वेबसाइट पर React जोड़ें
```
`जोड़ें` => `ऐड करें`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
 <Intro>
 
-React has been designed from the start for gradual adoption, and you can use as little or as much React as you need. Whether you're working with micro-frontends, an existing system, or just giving React a try, you can start adding interactive React components to an HTML page with just a few lines of code—and no build tooling!
+React को शुरू से ही धीरे-धीरे अपनाने के लिए डिज़ाइन किया गया है, और आप जितना चाहें उतना कम या ज्यादा React का उपयोग कर सकते हैं। चाहे आप माइक्रो-फ्रंटेंड, एक मौजूदा सिस्टम के साथ काम कर रहे हों, या सिर्फ React को आज़मा रहे हों, आप HTML पेज पर इंटरेक्टिव React कौम्पोनॅन्ट को कोड की कुछ पंक्तियों के साथ जोड़ना शुरू कर सकते हैं - और कोई बिल्ड टूलिंग नहीं!
```
`फ्रंटेंड, एक मौजूदा सिस्टम` => `फ्रंटेंड, या एक मौजूदा सिस्टम`

arshadkazmi42 replied:
`कौम्पोनॅन्ट को कोड` => `कौम्पोनॅन्ट के कोड`

arshadkazmi42 replied:
`जोड़ना` => `ऐड करना`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
 <Intro>
 
-React has been designed from the start for gradual adoption, and you can use as little or as much React as you need. Whether you're working with micro-frontends, an existing system, or just giving React a try, you can start adding interactive React components to an HTML page with just a few lines of code—and no build tooling!
+React को शुरू से ही धीरे-धीरे अपनाने के लिए डिज़ाइन किया गया है, और आप जितना चाहें उतना कम या ज्यादा React का उपयोग कर सकते हैं। चाहे आप माइक्रो-फ्रंटेंड, एक मौजूदा सिस्टम के साथ काम कर रहे हों, या सिर्फ React को आज़मा रहे हों, आप HTML पेज पर इंटरेक्टिव React कौम्पोनॅन्ट को कोड की कुछ पंक्तियों के साथ जोड़ना शुरू कर सकते हैं - और कोई बिल्ड टूलिंग नहीं!
```
`और कोई बिल्ड टूलिंग नहीं!` => `और कोई बिल्ड टूलिंग की ज़रूरत नहीं!`

I am not very satisfied with this suggestion also. Let me know if you have any other suggestion for this

Anishpras replied:
I think your suggestion is right as we need to express our stress on that part.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
 </Intro>
 
-## Add React in one minute {/*add-react-in-one-minute*/}
+## एक मिनट में React जोड़ें {/*add-react-in-one-minute*/}
```
`जोड़ें` => `ऐड करें`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+## एक मिनट में React जोड़ें {/*add-react-in-one-minute*/}
 
-You can add a React component to an existing HTML page in under a minute. Try this out with your own website or [an empty HTML file](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip)—all you need is an internet connection and a text editor like Notepad (or VSCode—check out our guide on [how to set yours up](/learn/editor-se ...
+आप एक मिनट से भी कम समय में मौजूदा HTML पृष्ठ पर एक React कौम्पोनॅन्ट जोड़ सकते हैं। इसे अपनी वेबसाइट या [एक खाली HTML फ़ाइल](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip) के साथ आज़माएं—आपको बस एक इंटरनेट कनेक्शन और नोटपैड (या VSCode—को [कैसे सेट अप करें](/learn/editor-setup/) इस पर हमारी मार्गदर्शिका देखें) जैसे टेक्स ...
```
`पृष्ठ` => `पेज`

arshadkazmi42 replied:
@Anishpras I think you missed this

I think you can do find and replace all for this

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+## एक मिनट में React जोड़ें {/*add-react-in-one-minute*/}
 
-You can add a React component to an existing HTML page in under a minute. Try this out with your own website or [an empty HTML file](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip)—all you need is an internet connection and a text editor like Notepad (or VSCode—check out our guide on [how to set yours up](/learn/editor-se ...
+आप एक मिनट से भी कम समय में मौजूदा HTML पृष्ठ पर एक React कौम्पोनॅन्ट जोड़ सकते हैं। इसे अपनी वेबसाइट या [एक खाली HTML फ़ाइल](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip) के साथ आज़माएं—आपको बस एक इंटरनेट कनेक्शन और नोटपैड (या VSCode—को [कैसे सेट अप करें](/learn/editor-setup/) इस पर हमारी मार्गदर्शिका देखें) जैसे टेक्स ...
```
`जोड़` => `ऐड`

arshadkazmi42 replied:
@Anishpras This is also missed

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+## एक मिनट में React जोड़ें {/*add-react-in-one-minute*/}
 
-You can add a React component to an existing HTML page in under a minute. Try this out with your own website or [an empty HTML file](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip)—all you need is an internet connection and a text editor like Notepad (or VSCode—check out our guide on [how to set yours up](/learn/editor-se ...
+आप एक मिनट से भी कम समय में मौजूदा HTML पृष्ठ पर एक React कौम्पोनॅन्ट जोड़ सकते हैं। इसे अपनी वेबसाइट या [एक खाली HTML फ़ाइल](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip) के साथ आज़माएं—आपको बस एक इंटरनेट कनेक्शन और नोटपैड (या VSCode—को [कैसे सेट अप करें](/learn/editor-setup/) इस पर हमारी मार्गदर्शिका देखें) जैसे टेक्स ...
```
`मार्गदर्शिका` => `गाइड`

arshadkazmi42 replied:
@Anishpras This is also missed

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+आप एक मिनट से भी कम समय में मौजूदा HTML पृष्ठ पर एक React कौम्पोनॅन्ट जोड़ सकते हैं। इसे अपनी वेबसाइट या [एक खाली HTML फ़ाइल](https://gist.github.com/rachelnabors/7b33305bf33776354797a2e3c1445186/archive/859eac2f7079c9e1f0a6eb818a9684a464064d80.zip) के साथ आज़माएं—आपको बस एक इंटरनेट कनेक्शन और नोटपैड (या VSCode—को [कैसे सेट अप करें](/learn/editor-setup/) इस पर हमारी मार्गदर्शिका देखें) जैसे टेक्स ...
 
-### Step 1: Add an element to the HTML {/*step-1-add-an-element-to-the-html*/}
+### चरण 1: HTML में एक एलिमेंट जोड़ें {/*step-1-add-an-element-to-the-html*/}
```
`चरण` => `स्टेप`

I feel this will be better. Let me know your thoughts

arshadkazmi42 replied:
`जोड़ें` => `ऐड करें`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+### चरण 1: HTML में एक एलिमेंट जोड़ें {/*step-1-add-an-element-to-the-html*/}
 
-In the HTML page you want to edit, add an HTML element like an empty `<div>` tag with a unique `id` to mark the spot where you want to display something with React.
+जिस HTML पृष्ठ में आप संपादित करना चाहते हैं, जहां आप React के साथ कुछ प्रदर्शित करना चाहते हैं, उस स्थान पर एक खाली `<div>` टैग के साथ एक अनोखा ‘id’ वाला HTML एलिमेंट जोड़ें।
```
```
जिस HTML पृष्ठ में आप संपादित करना चाहते हैं, जहां आप React के साथ कुछ प्रदर्शित करना चाहते हैं, उस स्थान पर एक खाली `<div>` टैग के साथ एक अनोखा ‘id’ वाला HTML एलिमेंट जोड़ें।
``` 
=> 
```
जिस HTML पेज को एडिट करके आप React के साथ कुछ डिस्प्ले करना चाहते हैं, वहां पर  एक खाली `<div>` टैग के साथ एक यूनिक ‘id’ वाला HTML एलिमेंट ऐड करें।
```

I have restructured complete line, let me know if you have any other inputs on this

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+जिस HTML पृष्ठ में आप संपादित करना चाहते हैं, जहां आप React के साथ कुछ प्रदर्शित करना चाहते हैं, उस स्थान पर एक खाली `<div>` टैग के साथ एक अनोखा ‘id’ वाला HTML एलिमेंट जोड़ें।
 
-You can place a "container" element like this `<div>` anywhere inside the `<body>` tag. React will replace any existing content inside HTML elements, so they are usually empty. You can have as many of these HTML elements on one page as you need.
+आप इस तरह का "कंटेनर" एलिमेंट `<div>` को `<body>` टैग के अंदर कहीं भी रख सकते हैं। React HTML एलिमेंट के अंदर किसी भी मौजूदा अंतर्वस्तु को बदल देगा, इसलिए वे आमतौर पर खाली होते हैं। आपके पास इन HTML एलिमेंट्स में से एक पृष्ठ पर जितनी आवश्यकता हो उतनी हो सकती है।
```
`अंतर्वस्तु` => `कंटेंट`

arshadkazmi42 replied:
`आपके पास इन HTML एलिमेंट्स में से एक पृष्ठ पर जितनी आवश्यकता हो उतनी हो सकती है।` => `आप अपनी ज़रूरत के हिसाब से ऐसे कितने भी HTML एलिमेंट्स एक पेज पर इस्तेमाल कर सकते हैं।`

arshadkazmi42 reviewed (changes requested):
Thank you for working on this.

I have added first round of review till line 28.

Lets close all the comments, then we can proceed further on rest of the page review.

You can also read about our review process here #23

Anishpras commented:
@arshadkazmi42  I have changed as you requested and replaced a few common words. Thank you for your valuable feedback.

arshadkazmi42 commented:
@Anishpras thank you for quickly working on this.

I think you missed comments added on line 13 you can find those at this link https://github.com/reactjs/hi.reactjs.org/pull/172/files#r780633931

There are 3 comments on that line. Can we get those also fixed.

Anishpras commented:
> @Anishpras thank you for quickly working on this.
> 
> I think you missed comments added on line 13 you can find those at this link https://github.com/reactjs/hi.reactjs.org/pull/172/files#r780633931
> 
> There are 3 comments on that line. Can we get those also fixed.

@arshadkazmi42  I have fixed those lines.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -1,56 +1,56 @@
 ```
 
-### Step 2: Add the Script Tags {/*step-2-add-the-script-tags*/}
+### स्टेप 2: script टैग ऐड करें {/*step-2-add-the-script-tags*/}
````
`script` => `Script`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+### स्टेप 2: script टैग ऐड करें {/*step-2-add-the-script-tags*/}
 
-In the HTML page, right before the closing `</body>` tag, add three `<script>` tags for the following files:
+HTML पेज में, समापन `</body>` टैग से ठीक पहले, निम्न फ़ाइलों के लिए तीन `<script>` टैग ऐड करें:
```
`समापन` => `क्लोजिंग`

how about this?

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
-- [**react.development.js**](https://unpkg.com/react@17/umd/react.development.js) loads the core of React
-- [**react-dom.development.js**](https://unpkg.com/react-dom@17/umd/react-dom.development.js) lets React render HTML elements to the [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model).
-- **like_button.js** is where you'll write your component in step 3!
+- [**react.development.js**](https://unpkg.com/react@17/umd/react.development.js) React के मूल को लोड करता है
```
`मूल` => `कोर`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
-- [**react-dom.development.js**](https://unpkg.com/react-dom@17/umd/react-dom.development.js) lets React render HTML elements to the [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model).
-- **like_button.js** is where you'll write your component in step 3!
+- [**react.development.js**](https://unpkg.com/react@17/umd/react.development.js) React के मूल को लोड करता है
+- [**react-dom.development.js**](https://unpkg.com/react-dom@17/umd/react-dom.development.js) React को HTML एलिमेंट को प्रस्तुत करने देता है [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model).
```
```
React को HTML एलिमेंट को प्रस्तुत करने देता है [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model).
```
=>
```
React को HTML एलिमेंट्स [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model) को रेंडर करने देता है। 
```

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
 <Gotcha>
 
-When deploying, replace "development.js" with "production.min.js".
+परिनियोजित करते समय, "Development.js" को "production.min.js" से बदलें।
```
`परिनियोजित` => `डेप्लॉय`

arshadkazmi42 replied:
`"Development.js"` => `"development.js"`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -1,56 +1,56 @@
+### स्टेप 3: एक React कौम्पोनॅन्ट बनाएँ {/*step-3-create-a-react-component*/}
 
-Create a file called **like_button.js** next to your HTML page, add this code snippet, and save the file. This code defines a React component called `LikeButton`. [You can learn more about making components in our guides.](/learn/your-first-component)
+अपने HTML पेज के बगल में **like_button.js** नाम की एक फाइल बनाएं, इस कोड स्निपेट को ऐड करें और फाइल को सेव करें। यह कोड `LikeButton` नामक एक React कौम्पोनॅन्ट को परिभाषित करता है। [आप हमारे गाइड में कौम्पोनॅन्ट बनाने के बारे में अधिक जान सकते हैं।](/learn/your-first-component)
```
`परिभाषित` => `डिफाइन`

Not very sure about this. Do you have any other alternatives here? `परिभाषित` sounds a bit complex.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
 
   if (liked) {
-    return 'You liked this!';
+    return 'आपको यह पसंद आया!';
```
Let's not translate any part of code except the comments

`आपको यह पसंद आया!` => `You liked this`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
       onClick: () => setLiked(true),
     },
-    'Like'
+    'पसंद करें'
```
`पसंद करें` => `Like`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
+### स्टेप 4: पेज पर अपना React कौम्पोनॅन्ट ऐड करें {/*step-4-add-your-react-component-to-the-page*/}
 
-Lastly, add two lines to the bottom of **like_button.js**. These two lines of code find the `<div>` you added to your HTML in the first step and then display the "Like" button React component inside of it.
+अंत में, **like_button.js** के नीचे दो पंक्तियाँ ऐड करें। कोड की ये दो पंक्तियाँ पहले स्टेप में आपके द्वारा अपने HTML में ऐडे गए `<div>` को ढूंढती हैं और फिर उसके अंदर "पसंद करें" बटन React कौम्पोनॅन्ट प्रदर्शित करती हैं।
```
`ऐडे` => `ऐड किये गए`

arshadkazmi42 replied:
@Anishpras here `गए` is added two times at line 77. Can you update that. seems a typo from my end which added the suggestion

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
+### स्टेप 4: पेज पर अपना React कौम्पोनॅन्ट ऐड करें {/*step-4-add-your-react-component-to-the-page*/}
 
-Lastly, add two lines to the bottom of **like_button.js**. These two lines of code find the `<div>` you added to your HTML in the first step and then display the "Like" button React component inside of it.
+अंत में, **like_button.js** के नीचे दो पंक्तियाँ ऐड करें। कोड की ये दो पंक्तियाँ पहले स्टेप में आपके द्वारा अपने HTML में ऐडे गए `<div>` को ढूंढती हैं और फिर उसके अंदर "पसंद करें" बटन React कौम्पोनॅन्ट प्रदर्शित करती हैं।
```
`"पसंद करें"` => `"Like"`

arshadkazmi42 replied:
`प्रदर्शित` => `को डिस्प्ले`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -59,216 +59,218 @@ function LikeButton() {
 ```
 
-**Congratulations! You have just rendered your first React component to your website!**
+**बधाई हो! आपने अभी-अभी अपनी वेबसाइट पर अपना पहला React कौम्पोनॅन्ट प्रस्तुत किया है!**
````
`प्रस्तुत` => `रेंडर`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
 
-- [View the full example source code](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9)
-- [Download the full example (2KB zipped)](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9/archive/7b41a88cb1027c9b5d8c6aff5212ecd3d0493504.zip)
+- [पूरा उदाहरण स्रोत कोड देखें](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9)
```
`स्रोत` => `सोर्स`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -59,216 +59,218 @@ function LikeButton() {
-- [View the full example source code](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9)
-- [Download the full example (2KB zipped)](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9/archive/7b41a88cb1027c9b5d8c6aff5212ecd3d0493504.zip)
+- [पूरा उदाहरण स्रोत कोड देखें](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9)
+- [पूरा उदाहरण डाउनलोड करें (2KB ज़िपित)](https://gist.github.com/rachelnabors/c64b3aeace8a191cf5ea6fb5202e66c9/archive/7b41a88cb1027c9b5d8c6aff5212ecd3d0493504.zip)
```
`zipped`  => `ज़ीपड`

how about this? Or we can leave it as is in english

Anishpras replied:
I think English will be better.

arshadkazmi42 reviewed (changes requested):
Thank you for fixes.

Added review for next part of the file till line 88.

Once these are done, I will continue the review for next section.

Anishpras commented:
@arshadkazmi42 I have done till line number 87.

arshadkazmi42 commented:
@Anishpras  There was a typo I have unresolved that comment its at line 77 can you check that. `गए गए` is added two times

Anishpras commented:
> @Anishpras There was a typo I have unresolved that comment its at line 77 can you check that. `गए गए` is added two times

@arshadkazmi42 I have removed the redundant words.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+यहाँ [JSX के साथ एक उदाहरण HTML फ़ाइल](https://raw.githubusercontent.com/reactjs/reactjs.org/main/static/html/single-file-example.html) है जिसे आप डाउनलोड कर सकते हैं और उसके साथ खेल सकते हैं।
 
-This approach is fine for learning and creating simple demos. However, it makes your website slow and **isn't suitable for production**. When you're ready to move forward, remove this new `<script>` tag and the `type="text/babel"` attributes you've added. Instead, in the next section you will set up a JSX preprocessor to convert all your `<script>` tags automatically.
+सरल डेमो सीखने और बनाने के लिए यह दृष्टिकोण ठीक है। हालांकि, यह आपकी वेबसाइट को धीमा कर देता है और **प्रोडक्शन के लिए उपयुक्त नहीं है**। जब आप आगे बढ़ने के लिए तैयार हों, तो यह नया `<script>` टैग और आपके द्वारा ऐडी गई `type="text/babel"` एट्रिब्यूट को हटा दें। इसके बजाय, अगले भाग में आप अपने सभी `<script>` टैग को स्वचालित रूप से बदलने के लिए एक JSX प्रीप्रोसेसर स्थापित करेंगे।
```
`दृष्टिकोण` => `तरीका`

arshadkazmi42 replied:
`ऐडी` => `ऐड की`

arshadkazmi42 replied:
`स्थापित` => `सेटअप`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+### किसी प्रोजेक्ट में JSX ऐड करें {/*add-jsx-to-a-project*/}
 
-Adding JSX to a project doesn't require complicated tools like a [bundler](/learn/start-a-new-react-project#custom-toolchains) or a development server. Adding a JSX preprocessor is a lot like adding a CSS preprocessor.
+किसी प्रोजेक्ट में JSX को ऐडने के लिए [बंडलर] (/learn/start-a-new-react-project#custom-toolchains) या एक डेवलपमेंट सर्वर जैसे जटिल टूल की आवश्यकता नहीं होती है। JSX प्रीप्रोसेसर ऐडना एक CSS प्रीप्रोसेसर ऐडने जैसा है।
```
`[बंडलर] (/learn/start-a-new-react-project#custom-toolchains) या एक डेवलपमेंट सर्वर जैसे जटिल टूल की आवश्यकता नहीं होती है।` => `[bundler](/learn/start-a-new-react-project#custom-toolchains) जैसे जटिल टूल या एक डेवलपमेंट सर्वर की आवश्यकता नहीं होती है।`

remove extra space

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+### किसी प्रोजेक्ट में JSX ऐड करें {/*add-jsx-to-a-project*/}
 
-Adding JSX to a project doesn't require complicated tools like a [bundler](/learn/start-a-new-react-project#custom-toolchains) or a development server. Adding a JSX preprocessor is a lot like adding a CSS preprocessor.
+किसी प्रोजेक्ट में JSX को ऐडने के लिए [बंडलर] (/learn/start-a-new-react-project#custom-toolchains) या एक डेवलपमेंट सर्वर जैसे जटिल टूल की आवश्यकता नहीं होती है। JSX प्रीप्रोसेसर ऐडना एक CSS प्रीप्रोसेसर ऐडने जैसा है।
```
`ऐडना` => `ऐड करना`

arshadkazmi42 replied:
@Anishpras I think you removed this part of the line `JSX प्रीप्रोसेसर ऐड करना एक CSS प्रीप्रोसेसर ऐड करने जैसा है।`

Can you add it back at line 197

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+### किसी प्रोजेक्ट में JSX ऐड करें {/*add-jsx-to-a-project*/}
 
-Adding JSX to a project doesn't require complicated tools like a [bundler](/learn/start-a-new-react-project#custom-toolchains) or a development server. Adding a JSX preprocessor is a lot like adding a CSS preprocessor.
+किसी प्रोजेक्ट में JSX को ऐडने के लिए [बंडलर] (/learn/start-a-new-react-project#custom-toolchains) या एक डेवलपमेंट सर्वर जैसे जटिल टूल की आवश्यकता नहीं होती है। JSX प्रीप्रोसेसर ऐडना एक CSS प्रीप्रोसेसर ऐडने जैसा है।
```
`ऐडने` => `ऐड करने`

arshadkazmi42 replied:
same as above

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+किसी प्रोजेक्ट में JSX को ऐडने के लिए [बंडलर] (/learn/start-a-new-react-project#custom-toolchains) या एक डेवलपमेंट सर्वर जैसे जटिल टूल की आवश्यकता नहीं होती है। JSX प्रीप्रोसेसर ऐडना एक CSS प्रीप्रोसेसर ऐडने जैसा है।
 
-Go to your project folder in the terminal, and paste these two commands (**Be sure you have [Node.js](https://nodejs.org/) installed!**):
+टर्मिनल में अपने प्रोजेक्ट फ़ोल्डर में जाएं, और इन दो आदेशों को पेस्ट करें (**सुनिश्चित करें कि आपने [Node.js](https://nodejs.org/) इंस्टॉल किया हुआ है!**):
```
`आदेशों` => `कमांड्स`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+टर्मिनल में अपने प्रोजेक्ट फ़ोल्डर में जाएं, और इन दो आदेशों को पेस्ट करें (**सुनिश्चित करें कि आपने [Node.js](https://nodejs.org/) इंस्टॉल किया हुआ है!**):
 
-1. `npm init -y` (if it fails, [here's a fix](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))
+1. `npm init -y` (यदि यह विफल रहता है, [यहाँ एक समाधान है](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))
```
`विफल रहता` => `फ़ैल होता`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -182,93 +184,93 @@ with:
+JSX प्रीप्रोसेसर को स्थापित करने के लिए आपको केवल npm की आवश्यकता है। आपको किसी और चीज के लिए इसकी आवश्यकता नहीं होगी। React और एप्लिकेशन कोड दोनों बिना किसी बदलाव के `<script>` टैग के रूप में रह सकते हैं।
 
-Congratulations! You just added a **production-ready JSX setup** to your project.
+बधाई हो! आपने अभी-अभी अपने प्रोजेक्ट में **प्रोडक्शन के लिए तैयार JSX सेटअप** ऐडा है।
```
`ऐडा` => `ऐड किया`

arshadkazmi42 replied:
@Anishpras seems you missed this.

arshadkazmi42 reviewed (changes requested):
Great going.

Added comments till line 209

Anishpras commented:
@arshadkazmi42 I have updated till 209.

arshadkazmi42 commented:
@Anishpras You missed one and at one line of part got deleted, I have marked those two as unresolved. can you resolved those before continuing further

Anishpras commented:
@arshadkazmi42 Sorry. I have resolved those lines.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
+### JSX प्रीप्रोसेसर चलाएँ {/*run-the-jsx-preprocessor*/}
 
-You can preprocess JSX so that every time you save a file with JSX in it, the transform will be re-run, converting the JSX file into a new, plain JavaScript file.
+आप JSX को प्रीप्रोसेस कर सकते हैं ताकि हर बार जब आप किसी फ़ाइल को JSX के साथ सहेजते हैं, तो JSX फ़ाइल को एक नई, सादा जावास्क्रिप्ट फ़ाइल में परिवर्तित करते हुए, ट्रांसफ़ॉर्म फिर से चलाया जाएगा।
```
`सादा` => `सादी`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
-2. In your terminal, run this command: `npx babel --watch src --out-dir . --presets react-app/prod ` (Don't wait for it to finish! This command starts an automated watcher for JSX.)
-3. Move your JSX-ified **like_button.js** to the new **src** folder (or create a **like_button.js** containing this [JSX starter code](https://gist.githubusercontent.com/rachelnabors/ffbc9a0e33665a58d4cfdd1676f05453/raw/652003ff54d2dab8a1a1e5cb3bb1e28ff207c1a6/like_button.js))
+1. **src** नाम का फोल्डर बनाएं
+2. अपने टर्मिनल में, यह कमांड चलाएँ: `npx babel --watch src --out-dir । --presets React-app/prod ` (इसके खत्म होने की प्रतीक्षा न करें! यह कमांड JSX के लिए एक स्वचालित वॉचर शुरू करता है।)
```
```
`npx babel --watch src --out-dir । --presets React-app/prod `
```
=>
```
`npx babel --watch src --out-dir . --presets react-app/prod `
```

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
-2. In your terminal, run this command: `npx babel --watch src --out-dir . --presets react-app/prod ` (Don't wait for it to finish! This command starts an automated watcher for JSX.)
-3. Move your JSX-ified **like_button.js** to the new **src** folder (or create a **like_button.js** containing this [JSX starter code](https://gist.githubusercontent.com/rachelnabors/ffbc9a0e33665a58d4cfdd1676f05453/raw/652003ff54d2dab8a1a1e5cb3bb1e28ff207c1a6/like_button.js))
+1. **src** नाम का फोल्डर बनाएं
+2. अपने टर्मिनल में, यह कमांड चलाएँ: `npx babel --watch src --out-dir । --presets React-app/prod ` (इसके खत्म होने की प्रतीक्षा न करें! यह कमांड JSX के लिए एक स्वचालित वॉचर शुरू करता है।)
```
`स्वचालित` => `ऑटोमेटेड`

Anishpras replied:
I have changed it at line 193 also.

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 <Gotcha>
 
-If you see an error message saying "You have mistakenly installed the `babel` package", you might have missed [the previous step](#add-jsx-to-a-project). Perform it in the same folder, and then try again.
+यदि आपको यह कहते हुए एक त्रुटि संदेश दिखाई देता है कि "आपने गलती से `babel` पैकेज सेटअप कर लिया है", तो आप [पिछला स्टेप](#add-jsx-to-a-project) चूक गए होंगे। इसे उसी फ़ोल्डर में निष्पादित करें, और फिर पुन: प्रयास करें।
```
`त्रुटि संदेश` => `एरर मैसेज`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 </Gotcha>
 
-As a bonus, this also lets you use modern JavaScript syntax features like classes without worrying about breaking older browsers. The tool we just used is called Babel, and you can learn more about it from [its documentation](https://babeljs.io/docs/en/babel-cli/).
+एक बोनस के रूप में, यह आपको पुराने ब्राउज़रों को तोड़ने की चिंता किए बिना classes जैसी आधुनिक जावास्क्रिप्ट सिंटैक्स सुविधाओं का उपयोग करने देता है। हमारे द्वारा अभी-अभी उपयोग किए गए टूल को babel कहा जाता है, और आप इसके बारे में [इसके दस्तावेज़ीकरण](https://babeljs.io/docs/en/babel-cli/) से अधिक जान सकते हैं।
```
`को तोड़ने` => `में ब्रेक होने`

arshadkazmi42 replied:
`दस्तावेज़ीकरण` => `डॉक्यूमेंटेशन`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
+एक बोनस के रूप में, यह आपको पुराने ब्राउज़रों को तोड़ने की चिंता किए बिना classes जैसी आधुनिक जावास्क्रिप्ट सिंटैक्स सुविधाओं का उपयोग करने देता है। हमारे द्वारा अभी-अभी उपयोग किए गए टूल को babel कहा जाता है, और आप इसके बारे में [इसके दस्तावेज़ीकरण](https://babeljs.io/docs/en/babel-cli/) से अधिक जान सकते हैं।
 
-If you're getting comfortable with build tools and want them to do more for you, [we cover some of the most popular and approachable toolchains here](/learn/start-a-new-react-project).
+यदि आप बिल्ड टूल्स के साथ सहज हो रहे हैं और चाहते हैं कि वे आपके लिए और अधिक करें, [हम यहां कुछ सबसे लोकप्रिय और पहुंचने योग्य टूलचेन को कवर करते हैं](/learn/start-a-new-react-project)।
```
`सहज` => `कम्फ़र्टेबल`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
+<DeepDive title="React बिना JSX के">
 
-Originally JSX was introduced to make writing components with React feel as familiar as writing HTML. Since then, the syntax has become widespread. However, there may be instances where you do not want to use or cannot use JSX. You have two options:
+कोर रूप से JSX को React के साथ लेखन कौम्पोनॅन्ट को HTML लिखने के रूप में परिचित बनाने के लिए पेश किया गया था। तब से, वाक्यविन्यास व्यापक हो गया है। हालाँकि, ऐसे उदाहरण हो सकते हैं जहाँ आप JSX का उपयोग नहीं करना चाहते हैं या नहीं कर सकते हैं। आपके पास दो विकल्प हैं:
```
`के साथ लेखन कौम्पोनॅन्ट` => `के कौम्पोनॅन्ट लिखने`

arshadkazmi42 replied:
`विकल्प` => `ऑप्शन`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 
-- Use a JSX alternative like [htm](https://github.com/developit/htm) which doesn't use a compiler—it uses JavaScript's native Tagged Templates.
-- Use [`React.createElement()`](/reference/createelement), which has a special structure explained below.
+- [htm](https://github.com/developit/htm) जैसे JSX विकल्प का उपयोग करें, जो कंपाइलर का उपयोग नहीं करता है—यह जावास्क्रिप्ट के कोर टैग किए गए टेम्प्लेट का उपयोग करता है।
```
`विकल्प` => `अल्टरनेटिव`

arshadkazmi42 replied:
`कोर` => `नेटिव`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
-- Use a JSX alternative like [htm](https://github.com/developit/htm) which doesn't use a compiler—it uses JavaScript's native Tagged Templates.
-- Use [`React.createElement()`](/reference/createelement), which has a special structure explained below.
+- [htm](https://github.com/developit/htm) जैसे JSX विकल्प का उपयोग करें, जो कंपाइलर का उपयोग नहीं करता है—यह जावास्क्रिप्ट के कोर टैग किए गए टेम्प्लेट का उपयोग करता है।
+- [`React.createElement()`](/reference/createelement) का उपयोग करें, जिसकी एक विशेष संरचना नीचे बताई गई है।
```
`संरचना` => `स्ट्रक्चर`

arshadkazmi42 replied:
`बताई गई` => `बताया गया`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
+- [`React.createElement()`](/reference/createelement) का उपयोग करें, जिसकी एक विशेष संरचना नीचे बताई गई है।
 
-With JSX, you would write a component like so:
+JSX के साथ, आप एक कौम्पोनॅन्ट लिखेंगे जैसे::
```
`जैसे::` => `जैसे:`

remove extra colon

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -176,99 +176,99 @@ return React.createElement(
 ```jsx
 function Hello(props) {
-  return <div>Hello {props.toWhat}</div>;
+  return <div>नमस्ते {props.toWhat}</div>;
````
`return <div>नमस्ते {props.toWhat}</div>;` => `return <div>Hello {props.toWhat}</div>;`

lets not change codes

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 }
 
-ReactDOM.render(<Hello toWhat="World" />, document.getElementById('root'));
+ReactDOM.render(<Hello toWhat="दुनिया" />, document.getElementById('root'));
```
`दुनिया` => `World`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -176,99 +176,99 @@ return React.createElement(
 ```js
 function Hello(props) {
-  return React.createElement('div', null, `Hello ${props.toWhat}`);
+  return React.createElement('div', null, `नमस्ते ${props.toWhat}`);
````
```
return React.createElement('div', null, `नमस्ते ${props.toWhat}`);
``` 
=> 
```
return React.createElement('div', null, `Hello ${props.toWhat}`);
```

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 
 ReactDOM.render(
-  React.createElement(Hello, {toWhat: 'World'}, null),
+  React.createElement(Hello, {toWhat: 'दुनिया'}, null),
```
`React.createElement(Hello, {toWhat: 'दुनिया'}, null),` => `React.createElement(Hello, {toWhat: 'World'}, null),`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -176,99 +176,99 @@ return React.createElement(
 ```
 
-It accepts three arguments: `React.createElement(component, props, children)`. Here's how they work:
+यह तीन तर्कों को स्वीकार करता है: `React.createElement(component, props, children)`। यहां बताया गया है कि वे कैसे काम करते हैं:
````
`तर्कों` => `आर्ग्यूमेंट्स`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
-1. A **component**, which can be a string representing an HTML element or a function component
-2. An object of any [**props** you want to pass](/learn/passing-props-to-a-component)
-3. An object of any **children** the component might have, such as text strings
+1. एक **component**, जो एक HTML एलिमेंट या एक फ़ंक्शन कौम्पोनॅन्ट का प्रतिनिधित्व करने वाला एक स्ट्रिंग हो सकता है
```
`का प्रतिनिधित्व` => `को रिप्रेजेंट`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
+3. कौम्पोनॅन्ट में किसी भी **children** का ऑब्जेक्ट हो सकता है, जैसे टेक्स्ट स्ट्रिंग्स
 
-If you get tired of typing `React.createElement()`, one common pattern is to assign a shorthand:
+यदि आप `React.createElement ()` टाइप करते-करते थक गए हैं, तो एक सामान्य पैटर्न शॉर्टहैंड असाइन करना है:
```
`React.createElement ()` => `React.createElement()`

remove extra space

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
```diff
@@ -176,99 +176,99 @@ return React.createElement(
 const e = React.createElement;
 
-ReactDOM.render(e('div', null, 'Hello World'), document.getElementById('root'));
+ReactDOM.render(e('div', null, 'नमस्ते दुनिया'), document.getElementById('root'));
```
`'नमस्ते दुनिया'` => `'Hello World'`

arshadkazmi42 commented on beta/src/pages/learn/add-react-to-a-website.md:
````diff
@@ -176,99 +176,99 @@ return React.createElement(
 ```
 
-If you use this shorthand form for `React.createElement()`, it can be almost as convenient to use React without JSX.
+यदि आप `React.createElement ()` के लिए इस शॉर्टहैंड फॉर्म का उपयोग करते हैं, तो JSX के बिना React का उपयोग करना लगभग उतना ही सुविधाजनक हो सकता है।
````
`React.createElement ()` => `React.createElement()`

remove extra space

arshadkazmi42 reviewed (changes requested):
Add review for rest of the file. Once these are fixed its good to go from my end

Anishpras commented:
@arshadkazmi42 I have done all the changes. Thank you for the wonderful reviews.

arshadkazmi42 reviewed (approved):
Great work. :tada: 

All are resolved. Lets get this merged.

Anishpras commented:
> Great work. 🎉
> 
> All are resolved. Lets get this merged.

@arshadkazmi42 Hurray! Thank you for this seamless review process 🚀.
